src/Vector.cpp

```diff
@@ -0,0 +1,335 @@
+#include <Vector.h>
+#include <stdexcept>
+
+// ======================== c-tors & d-tor =======================
+
+/**
+ * ctor creates a vector with a given size and value
+ * @param size the size of the vector
+ * @param value the value to initialize the vector with
+ **/
+Vector::Vector(size_t size, int value):
+	m_size(size),
+	m_capacity(size),
+	m_vector(new int[size])
```

Comment:
Notice that you might be allocating a 0 sized array int. This essentially means that no memory is allocated and thus will require another allocation when an item is inserted (hurts performance). It is probably wiser to pre-allocate some memory in the array.

Reply:
fixed